### PR TITLE
Fixed asm code for func call emulation

### PR DESCRIPTION
**Pull request by sreznick**

### CODE REVIEW

**Comment by sreznick** [comment]
It was reason of smashing stack

**Comment by sreznick** [comment]
Comments contained good ideas of instructions but instructions did not fit them leading to damaged memory.

File: src/hotspot/cpu/riscv/vm_version_riscv.cpp [hunk 1]
```diff
@@ -781,14 +781,15 @@ void VM_Version::determine_features() {
...
-  a->addi(R2, R2, 16);    // addi  s0,sp,16
+  a->sd(R8, R2, 8);       // sd    s0,8(sp)
+  a->addi(R8, R2, 16);    // addi  s0,sp,16
   a->nop();               // nop
-  a->sd(R2, R8, 8);       // ld	  s0,8(sp)
-  a->sd(R2, R2, 16);      // addi  sp,sp,16
+  a->ld(R8, R2, 8);       // ld	  s0,8(sp)
+  a->addi(R2, R2, 16);    // addi  sp,sp,16
   a->ret();               // jr	  ra
-  uint32_t *code_end = (uint32_t *)a->pc();
```
Comment (AntonKozlov) [inline src/hotspot/cpu/riscv/vm_version_riscv.cpp:791]: Cannot compile without this line :(

**Review by AntonKozlov — APPROVED** [review]
Cool! I suppose that was hard to catch

File: src/hotspot/cpu/riscv/vm_version_riscv.cpp (hunk 1 above)
Comment (sreznick) [reply src/hotspot/cpu/riscv/vm_version_riscv.cpp:791]: Yes. My mistake preparing this PR

**Pull request merged**